_posts/2023-07-24-elytron-oidc-client-scope.adoc

```diff
@@ -0,0 +1,136 @@
+tags: oidc authentication 
+synopsis: How to secure a web application deployed to WildFly with OpenID Connect using additional scope values. 
+author: prarthonapaul
+---
+
+:toc: macro
+:toc-title:
+
+https://openid.net/developers/how-connect-works/[ OpenID Connect] is an identity layer on top of the OAuth 2.0 protocol that makes it possible for a client to verify a user’s identity based on authentication that’s performed by an OpenID provider. The https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest[OpenID Connect specifications] indicate that there are other scope values which may be included in the authentication request to ask permission to additional and specific resources.The purpose of this example is to demonstrate how to configure additional scope values to request access to additional fields.
+
+Note: This quickstart demonstates how to configure scope values in two different ways: 
+
+* Deployment Configuration
+
+* Subsystem Configuration
+
+toc::[]
+
+== Setting up Keycloak OpenID Provider
+
+It’s easy to set up Keycloak using Docker. Follow the steps in Keycloak’s https://www.keycloak.org/getting-started/getting-started-docker[ getting started guide ] to start Keycloak, create a realm called myrealm, create a user called myuser, and register a client called myclient.
+
+After registering our client, myclient, we also need to configure valid redirect URIs. Simply click on Clients and then on myclient. In the Valid Redirect URIs field, enter http://localhost:8090/simple-webapp-oidc/*.
+
+Note: The realm and client names are case sensitive. 
+
+== Server Configuration
+
+First clone the `elytron-example` repo locally: 
+```
+    git clone https://github.com/wildfly-security-incubator/elytron-examples
+    cd elytron-examples
+```
+
+We will be making use of two projects inside the repo: `elytron-oidc-client-scope` and `simple-webapp-oidc`. 
+
+In order to authenticate a user using OpenID connect, we need to configure the server to use OpenID attributes. This can be done using one of the two following configurations: 
+
+=== Deployment Configuration
+
+The configuration required to secure an application with OpenID Connect can be specified in the deployment.
+
+The first step is to create an `oidc.json` configuration file in the `WEB-INF` directory of the `simple-webapp-oidc` project. The following is the contents of the `oidc.json` file used in this example: 
+```
+{
+    "client-id" : "myclient",
+    "provider-url" : "${env.OIDC_PROVIDER_URL:http://localhost:8080}/realms/myrealm",
+    "public-client" : "true",
+    "scope" : "profile email phone address offline_access",
+    "principal-attribute" : "preferred_username",
+    "ssl-required" : "EXTERNAL"
+}
+```
+
+Note that we have specified the scope values inside the `oidc.json` file. The next step is to set the auth-method to OIDC in the application’s `web.xml` file by adding the following command: 
+```
+<login-config>
+    <auth-method>OIDC</auth-method>
```

Comment:
Same here.

Reply:
Implemented all changes till here. 
Still need to figure out how to use the bearer token and extract information obtained using the scope values. 
Had to change the file name to a current date, because CI was skipping it earlier since it was a future date. Will change the date back once it is ready to publish.

Reply:
All changes implemented. The updated version includes a modified `simple-webapp-oidc` that displays a list of claims obtained by configured scopes.